### PR TITLE
remove support for congruence closure computation

### PR DESCRIPTION
**Pull request by nikomatsakis**
Supporting congruence closure requires the `unioned_keys` iterator,
which in turn requires maintaining the parent/sibling pointers. This
grows the size of nodes, which in turn leads to a 2% slowdown in
various rustc test cases (e.g., the `coercions` benchmark in
perf.rust-lang.org). Given that this feature is not used in chalk or
anywhere else, no reason to support it.

Cargo bench measurements (3 runs each):

before
test unify::tests::big_array_bench_InPlace             ... bench:   1,335,817 ns/iter
test unify::tests::big_array_bench_InPlace             ... bench:   1,030,749 ns/iter
test unify::tests::big_array_bench_InPlace             ... bench:   1,339,391 ns/iter

after
test unify::tests::big_array_bench_InPlace             ... bench:     918,518 ns/iter
test unify::tests::big_array_bench_InPlace             ... bench:   1,190,575 ns/iter
test unify::tests::big_array_bench_InPlace             ... bench:     918,214 ns/iter

A bit uneven but clearly a win.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -23,9 +19,6 @@ extern crate log;
 #[cfg(feature = "persistent")]
 extern crate dogged;
 
-#[cfg(feature = "congruence-closure")]
-extern crate petgraph;
-
 pub mod snapshot_vec;
-pub mod cc;
 pub mod unify;
+
```
Comment (sgrif) [inline src/lib.rs]: ✂️

File: src/unify/mod.rs [hunk 2]
```diff
@@ -199,43 +197,31 @@ pub struct Snapshot<S: UnificationStore> {
...
+    fn redirect(&mut self, to: K) {
         self.parent = to;
-        self.sibling = sibling;
     }
 
-    fn root(&mut self, rank: u32, child: K, value: K::Value) {
+    fn root(&mut self, rank: u32, value: K::Value) {
         self.rank = rank;
-        self.child = child;
+        // self.child = child;
```
Comment (sgrif) [inline src/unify/mod.rs]: Maybe just kill it instead of commenting it out?

File: src/unify/mod.rs [hunk 3]
```diff
@@ -397,7 +370,7 @@ impl<S: UnificationStore> UnificationTable<S> {
         }
     }
 
-    /// Internal method to redirect `old_root_key` (which is currently
+    /// Internal method tox redirect `old_root_key` (which is currently
```
Comment (sgrif) [inline src/unify/mod.rs]: I don't think you meant to commit this

Comment (nikomatsakis) [reply src/unify/mod.rs]: you...are right

File: src/unify/mod.rs (hunk 2 above)
Comment (nikomatsakis) [reply src/unify/mod.rs]: oops

**Comment by nikomatsakis** [comment]
Grr travis you take too long

**Pull request merged**